File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        int l = speeds.length;
        // 남은 일 횟수
        // 7 3 9
        
        Queue<Integer> q = new LinkedList<>();
        for(int i = 0; i < l; i++){
            int rest = (100 - progresses[i]) / speeds[i];
            if((100 - progresses[i]) % speeds[i] != 0) rest++;
            
            q.offer(rest);
        }
        
        while(!q.isEmpty()){
            int now = q.poll();
            int num = 1;
            while(!q.isEmpty() && q.peek() <= now){
                q.poll();
                num++;
            }
            answer.add(num);
        }
        
        return answer.stream().mapToInt(i -> i).toArray();
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    temp = []
    l = int(len(progresses))
    i = 0
    while i < l :
        num = int((100 - progresses[i]) / speeds[i])
        if(int((100 - progresses[i]) % speeds[i]) > 0) : num += 1
        temp.append(num)
        i += 1

    # [7,3,9]
    now = temp[0]
    count = 1
    for n in temp[1:] :
        if(n <= now) :
            count += 1
        else :
            answer.append(count)
            now = n
            count = 1
    answer.append(count)
    return answer